src/Identifier/Normalizer/ChainIdentifierNormalizer.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Identifier\Normalizer;
+
+use ApiPlatform\Core\Exception\InvalidIdentifierException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Identifier normalizer.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class ChainIdentifierNormalizer implements IdentifierNormalizerInterface
+{
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
+    private $compositeIdentifierNormalizer;
+    private $identifierNormalizers;
+
+    public const HAS_IDENTIFIER_NORMALIZER = 'has_normalized_identifier';
+
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, array $identifierNormalizers)
+    {
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->compositeIdentifierNormalizer = new CompositeIdentifierNormalizer();
```

Comment:
Why not injecting it in the existing array?

Reply:
Because it's not implementing the IdentifierNormalizerInterface but it's called pre-normalization when we have more then 1 identifier. See  https://github.com/api-platform/core/pull/1478/files/b3037d9fa3ec663a8f1dd51c771e67a3aeab23f5#diff-9622bccedd195e39c3b3da920161ef42R69

Reply:
Why does it have a different behavior than other normalizers? It looks weird to me.